x-pack/plugins/endpoint/public/embeddables/resolver/models/indexed_process_tree.ts

```diff
@@ -15,21 +15,69 @@ import { levelOrder as baseLevelOrder } from '../lib/tree_sequencers';
 export function factory(processes: ResolverEvent[]): IndexedProcessTree {
   const idToChildren = new Map<string | undefined, ResolverEvent[]>();
   const idToValue = new Map<string, ResolverEvent>();
+  const idToAdjacent = new Map<string, AdjacentProcessMap>();
```

Comment:
Not sure I have a good solution or if it'd be too much of an overhaul but I wonder if we encapsulated these maps within a `Tree` class or something it would make it more clear how to access different nodes.

Reply:
@jonathan-buttner that's an interesting idea, but a little bit outside the scope of this PR: this map has a specific purpose to support what's needed for https://www.w3.org/WAI/PF/aria/roles#tree . If we did something more broad in the future, it could obviate the need for this.